chain/vm/fvm.go

```diff
@@ -303,6 +308,111 @@ func NewFVM(ctx context.Context, opts *VMOpts) (*FVM, error) {
 	}, nil
 }
 
+func NewDebugFVM(ctx context.Context, opts *VMOpts) (*FVM, error) {
+	state, err := state.LoadStateTree(cbor.NewCborStore(opts.Bstore), opts.StateBase)
+	if err != nil {
+		return nil, err
+	}
+
+	circToReport, err := opts.CircSupplyCalc(ctx, opts.Epoch, state)
+	if err != nil {
+		return nil, err
+	}
+
+	baseBstore := opts.Bstore
+	overlayBstore := blockstore.NewMemorySync()
+	cborStore := cbor.NewCborStore(overlayBstore)
+	vmBstore := blockstore.NewTieredBstore(overlayBstore, baseBstore)
+
+	fvmopts := &ffi.FVMOpts{
+		FVMVersion: 0,
+		Externs: &FvmExtern{
+			Rand:       opts.Rand,
+			Blockstore: vmBstore,
+			lbState:    opts.LookbackState,
+			base:       opts.StateBase,
+			epoch:      opts.Epoch,
+		},
+		Epoch:          opts.Epoch,
+		BaseFee:        opts.BaseFee,
+		BaseCircSupply: circToReport,
+		NetworkVersion: opts.NetworkVersion,
+		StateBase:      opts.StateBase,
+		Tracing:        EnableDetailedTracing,
+		Debug:          true,
+	}
```

Comment:
nit: we should probably have a helper function for this logic, so we don't duplicate it in multiple places. The only difference here is `Debug: true`, and the tiered blockstore. Ideally, we'd have a `DefaultFVMOpts(*VMOpts)` method, or something.

Reply:
seems reasonable, thanks!